src/components/button-toggle/button-toggle-group/button-toggle-group.test.tsx

```diff
@@ -0,0 +1,424 @@
+      label="Group Label"
+      onChange={() => {}}
+      inputWidth={50}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("group")).toHaveStyle({ width: "50%" });
+});
+
+test("should render with expected styles when labelInline prop is set", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      label="Group Label"
+      onChange={() => {}}
+      labelInline
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("group")).toHaveStyle({ flexWrap: "nowrap" });
+});
+
+test("should not render labelHelp or filedHelp when validationRedesignOptIn is true", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <ButtonToggleGroup
+        id="button-toggle-group-id"
+        label="Group Label"
+        labelHelp="Label Help"
+        fieldHelp="Field Help"
+        onChange={() => {}}
+      >
+        <ButtonToggle value="foo">Foo</ButtonToggle>
+        <ButtonToggle value="bar">Bar</ButtonToggle>
+      </ButtonToggleGroup>
+    </CarbonProvider>
+  );
+
+  expect(screen.queryByText("Label Help")).not.toBeInTheDocument();
+  expect(screen.queryByText("Field Help")).not.toBeInTheDocument();
+});
+
+test("should only allow the first button to be tabbable when none are selected", async () => {
+  const user = userEvent.setup();
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      onChange={() => {}}
+      value={undefined}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  // need await as tabindex conditions were not computed before assertion and test would fail
+  await user.tab();
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveFocus();
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveAttribute(
+    "tabindex",
+    "0"
+  );
+  expect(screen.getByRole("button", { name: "Bar" })).toHaveAttribute(
+    "tabindex",
+    "-1"
+  );
+});
+
+test("should only allow a selected button to be tabbable", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      onChange={() => {}}
+      value="bar"
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+      <ButtonToggle value="baz">Baz</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveAttribute(
+    "tabindex",
+    "-1"
+  );
+  expect(screen.getByRole("button", { name: "Bar" })).toHaveAttribute(
+    "tabindex",
+    "0"
+  );
+  expect(screen.getByRole("button", { name: "Baz" })).toHaveAttribute(
+    "tabindex",
+    "-1"
+  );
+});
+
+test("should focus on the previous button when the left arrow key is pressed", async () => {
```

Comment:
**suggestion**: to clarify the precondition that a button is already selected, what do you think of this adjustment?:

```suggestion
test("focuses the button left of the currently selected button, when the left arrow key is pressed", async () => {
```

Reply:
I think a mix of the above is fine, I think using `previous` is fine as the actual orientation isn't massively relevant and can be inferred anyway imo.